### ISSUE TITLE
Module generics confuse the compiler's understanding of types

### ISSUE THREAD

**Issue by IgorSamurovic** [wurstscript/WurstScript#594]
I have this module:
```
package MapModule
import TypeCasting
import HashMap

public module MapModule<T>
    protected static let zzz_map = new HashMap<T, thistype>()
    protected T mapRef 

    static function of(T t) returns thistype
        return zzz_map.get(t)

    function attachTo(T t)
        mapRef = t
        zzz_map.put(t, this)

    ondestroy
        zzz_map.remove(mapRef)
```

What it does is - it lets you attach an instance of a class to a T type object. This is what it looks like when used:

```
public class ResourceTypeStyle
    use MapModule<ResourceType>

    construct(ResourceType rt)
        attachTo(rt)
```

This won't even give you a warning as you type this code in. However, when you try to build the map, it says this:

> Error in File ResourceStyle.wurst line 6:
 Wrong parameter type when calling of.
Found ResourceType but expected ResourceType t
Error in File ResourceStyle.wurst line 41:
 Wrong parameter type when calling attachTo.
Found ResourceType but expected ResourceType t

This error doesn't even make any sense.

**Comment by Frotty** [comment]
has been addressed by 439535c09e75c581763ec1fd2da237e4dc7d4c11
also see #592 
Still should be properly done when refactoring.

**Comment by Frotty** [comment]
Seems to have been fixed in type-rework.
